src/Symfony/Component/Asset/VersionStrategy/RemoteJsonManifestVersionStrategy.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\VersionStrategy;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Reads the versioned path of an asset from a remote JSON manifest file.
+ *
+ * For example, the manifest file might look like this:
+ *     {
+ *         "main.js": "main.abc123.js",
+ *         "css/styles.css": "css/styles.555abc.css"
+ *     }
+ *
+ * You could then ask for the version of "main.js" or "css/styles.css".
+ */
+class RemoteJsonManifestVersionStrategy implements VersionStrategyInterface
+{
+    private $manifestData;
+    private $httpResponse;
+
+    /**
+     * @param string $manifestUrl Absolute url to the manifest file
+     */
+    public function __construct(string $manifestUrl, HttpClientInterface $httpClient)
+    {
+        $this->httpResponse = $httpClient->request('GET', $manifestUrl);
```

Comment:
clever, async = lazy :)
I'd suggest adding `['buffer' => true]` as option

Reply:
- 👍 for buffering.
- I need to check that an HTTP request is not fired on pages that don't use the asset package.
- I'll add the `accept: application/json` header, because that's what we expect.

Reply:
> I need to check that an HTTP request is not fired on pages that don't use the asset package.

one will be issued when this class is instantiated for sure.

Reply:
The HTTP request is issued even when using the command. That would be an issue.

I'm moving the request to the first call to `getManifestPath`.
If one needs to preload the manifest, it might be possible to make the request earlier with an other mecanism (a listener on `kernel.request`).